<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageWindow</class>
 <widget class="QMainWindow" name="ImageWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>图像处理</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>160</y>
      <width>1031</width>
      <height>511</height>
     </rect>
    </property>
    <property name="title">
     <string>原始图片信息</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>751</width>
       <height>451</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Panel</enum>
     </property>
     <property name="text">
      <string>请先选择要处理的图片</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="resize">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>50</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>860</x>
       <y>30</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>放大倍数：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>50</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>50</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>200%</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>110</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>90</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>图像锐化：</string>
     </property>
    </widget>
    <widget class="QSlider" name="sharpen">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>110</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>110</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>170</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>150</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>对比度：</string>
     </property>
    </widget>
    <widget class="QSlider" name="contrast">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>170</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>170</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>230</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>210</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>亮度：</string>
     </property>
    </widget>
    <widget class="QSlider" name="brightness">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>230</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>230</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>290</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>270</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>饱和度：</string>
     </property>
    </widget>
    <widget class="QSlider" name="saturation">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>290</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>256</number>
     </property>
     <property name="value">
      <number>128</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>290</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_33">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>400</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_34">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>380</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>平滑处理：</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_35">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>400</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="filter">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>400</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_36">
     <property name="geometry">
      <rect>
       <x>990</x>
       <y>350</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_37">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>330</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>色度：</string>
     </property>
    </widget>
    <widget class="QSlider" name="chroma">
     <property name="geometry">
      <rect>
       <x>820</x>
       <y>350</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-180</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_38">
     <property name="geometry">
      <rect>
       <x>794</x>
       <y>350</y>
       <width>20</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>90</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_2">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>150</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_3">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>200</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_4">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>260</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_5">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>320</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_6">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>380</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="reset">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>680</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>还原</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>680</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>保存</string>
    </property>
   </widget>
   <widget class="QPushButton" name="compare">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>680</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>对比</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadImage">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>680</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>选择图片</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>120</width>
      <height>721</height>
     </rect>
    </property>
    <property name="title">
     <string>工具栏</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>查看</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>画图</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>几何变换</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>颜色处理</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_7">
    <property name="title">
     <string>滤镜</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_8">
    <property name="title">
     <string>点运算</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_9">
    <property name="title">
     <string>频域处理</string>
    </property>
    <widget class="QMenu" name="menu_10">
     <property name="title">
      <string>低通滤波</string>
     </property>
     <addaction name="action_low_1"/>
     <addaction name="action_low_2"/>
     <addaction name="action_low_3"/>
     <addaction name="action_low_4"/>
     <addaction name="action_low_5"/>
    </widget>
    <widget class="QMenu" name="menu_11">
     <property name="title">
      <string>高通滤波</string>
     </property>
     <addaction name="action_high_1"/>
     <addaction name="action_high_2"/>
     <addaction name="action_high_3"/>
     <addaction name="action_high_4"/>
     <addaction name="action_high_5"/>
     <addaction name="action_high_6"/>
    </widget>
    <addaction name="actionFFT"/>
    <addaction name="actionIFFT"/>
    <addaction name="menu_10"/>
    <addaction name="menu_11"/>
   </widget>
   <widget class="QMenu" name="menu_12">
    <property name="title">
     <string>空域处理</string>
    </property>
    <widget class="QMenu" name="menua">
     <property name="title">
      <string>线性滤波</string>
     </property>
     <addaction name="actiona1"/>
     <addaction name="actiona2"/>
     <addaction name="actiona3"/>
     <addaction name="actiona4"/>
    </widget>
    <widget class="QMenu" name="menub">
     <property name="title">
      <string>非线性滤波</string>
     </property>
     <addaction name="actiona1_2"/>
     <addaction name="actiona2_2"/>
    </widget>
    <widget class="QMenu" name="menud">
     <property name="title">
      <string>图像锐化(边缘检测）</string>
     </property>
     <addaction name="action_grad"/>
     <addaction name="action_prewitt"/>
     <addaction name="action_sobel"/>
     <addaction name="action_log"/>
     <addaction name="action_qualc"/>
     <addaction name="action_lpls"/>
    </widget>
    <widget class="QMenu" name="menua_2">
     <property name="title">
      <string>直方图处理</string>
     </property>
     <widget class="QMenu" name="menu_6">
      <property name="title">
       <string>直方图规定化</string>
      </property>
      <addaction name="action_hisSML"/>
      <addaction name="action_hisGML"/>
     </widget>
     <addaction name="action_hisEqu"/>
     <addaction name="menu_6"/>
    </widget>
    <widget class="QMenu" name="menub_2">
     <property name="title">
      <string>灰度变换</string>
     </property>
     <addaction name="actionb1"/>
     <addaction name="actionn2"/>
     <addaction name="actionn3"/>
    </widget>
    <addaction name="menua"/>
    <addaction name="menub"/>
    <addaction name="menud"/>
    <addaction name="menua_2"/>
    <addaction name="menub_2"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_E"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_5"/>
   <addaction name="menu_7"/>
   <addaction name="menu_8"/>
   <addaction name="menu_9"/>
   <addaction name="menu_12"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionFFT">
   <property name="text">
    <string>FFT</string>
   </property>
  </action>
  <action name="actionIFFT">
   <property name="text">
    <string>IFFT</string>
   </property>
  </action>
  <action name="action_low_1">
   <property name="text">
    <string>理想低通滤波</string>
   </property>
  </action>
  <action name="action_low_2">
   <property name="text">
    <string>巴特沃斯低通滤波</string>
   </property>
  </action>
  <action name="action_low_3">
   <property name="text">
    <string>指数低通滤波</string>
   </property>
  </action>
  <action name="action_low_4">
   <property name="text">
    <string>梯形低通滤波</string>
   </property>
  </action>
  <action name="action_low_5">
   <property name="text">
    <string>高斯低通滤波</string>
   </property>
  </action>
  <action name="action_high_1">
   <property name="text">
    <string>理想高通滤波</string>
   </property>
  </action>
  <action name="action_high_2">
   <property name="text">
    <string>巴沃特斯高通滤波</string>
   </property>
  </action>
  <action name="action_high_3">
   <property name="text">
    <string>指数高通滤波</string>
   </property>
  </action>
  <action name="action_high_4">
   <property name="text">
    <string>梯形高通滤波</string>
   </property>
  </action>
  <action name="action_high_5">
   <property name="text">
    <string>高斯高通滤波</string>
   </property>
  </action>
  <action name="action_high_6">
   <property name="text">
    <string>高频增强滤波</string>
   </property>
  </action>
  <action name="actiona1">
   <property name="text">
    <string>邻域平均</string>
   </property>
  </action>
  <action name="actiona2">
   <property name="text">
    <string>选择平均</string>
   </property>
  </action>
  <action name="actiona3">
   <property name="text">
    <string>加权平均</string>
   </property>
  </action>
  <action name="actiona4">
   <property name="text">
    <string>Wiener滤波</string>
   </property>
  </action>
  <action name="actiona1_2">
   <property name="text">
    <string>中值滤波</string>
   </property>
  </action>
  <action name="actiona2_2">
   <property name="text">
    <string>序统计滤波</string>
   </property>
  </action>
  <action name="action_grad">
   <property name="text">
    <string>梯度法</string>
   </property>
  </action>
  <action name="action_prewitt">
   <property name="text">
    <string>Prewitt算子</string>
   </property>
  </action>
  <action name="action_sobel">
   <property name="text">
    <string>Sobel算子</string>
   </property>
  </action>
  <action name="action_log">
   <property name="text">
    <string>Log算子</string>
   </property>
  </action>
  <action name="action_qualc">
   <property name="text">
    <string>高通滤波</string>
   </property>
  </action>
  <action name="action_hisEqu">
   <property name="text">
    <string>直方图均衡化</string>
   </property>
  </action>
  <action name="actionb1">
   <property name="text">
    <string>比例线性变换</string>
   </property>
  </action>
  <action name="actionn2">
   <property name="text">
    <string>分段线性变换</string>
   </property>
  </action>
  <action name="actionn3">
   <property name="text">
    <string>非线性变换</string>
   </property>
  </action>
  <action name="action_hisSML">
   <property name="text">
    <string>单映射</string>
   </property>
  </action>
  <action name="action_hisGML">
   <property name="text">
    <string>组映射</string>
   </property>
  </action>
  <action name="action_lpls">
   <property name="text">
    <string>拉普拉斯</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
